src/model/tests/ModelUnitTest.cpp

```diff
@@ -94,23 +95,58 @@ void getRandomSubsetOfJoints(const Model & model,
+                                                       std::unordered_map<std::string, std::vector<double>>& removedJointPositions,
                                                        FreeFloatingPos& fullModelPos)
 {
     for(JointIndex jntIndex = 0; jntIndex < fullModel.getNrOfJoints(); jntIndex++)
     {
         // Check if joint is in reduced model
         std::string jointName = fullModel.getJointName(jntIndex);
 
-        // Only set non-zero position if the DOF size is exactly 1
         if (!isStringInVector(jointName, subsetOfJointsInReducedModel))
         {
-            if (fullModel.getJoint(jntIndex)->getNrOfDOFs() == 1)
+            size_t nrOfPosCoords = fullModel.getJoint(jntIndex)->getNrOfPosCoords();
+
+            if (nrOfPosCoords == 1)
             {
+                // For 1-DOF joints, set a random position
                 double jointConf = iDynTree::getRandomDouble();
-                removedJointPositions[jointName] = jointConf;
+                removedJointPositions[jointName] = {jointConf};
                 fullModelPos.jointPos()(fullModel.getJoint(jntIndex)->getPosCoordsOffset()) = jointConf;
             }
+            else if (nrOfPosCoords == 4)
+            {
+                // For spherical joints with quaternion parameterization, generate a random unit quaternion
+                double w = iDynTree::getRandomDouble();
+                double x = iDynTree::getRandomDouble();
+                double y = iDynTree::getRandomDouble();
+                double z = iDynTree::getRandomDouble();
+
+                // Normalize to unit quaternion
+                double norm = std::sqrt(w*w + x*x + y*y + z*z);
+                if (norm > 1e-12) {
+                    w /= norm; x /= norm; y /= norm; z /= norm;
+                } else {
+                    w = 1.0; x = 0.0; y = 0.0; z = 0.0;  // Default to identity
+                }
+
+                removedJointPositions[jointName] = {w, x, y, z};
+                fullModelPos.jointPos()(fullModel.getJoint(jntIndex)->getPosCoordsOffset() + 0) = w;
+                fullModelPos.jointPos()(fullModel.getJoint(jntIndex)->getPosCoordsOffset() + 1) = x;
+                fullModelPos.jointPos()(fullModel.getJoint(jntIndex)->getPosCoordsOffset() + 2) = y;
+                fullModelPos.jointPos()(fullModel.getJoint(jntIndex)->getPosCoordsOffset() + 3) = z;
+            }
+            else if (nrOfPosCoords > 1)
+            {
+                // For other multi-coordinate joints, set random values
+                std::vector<double> jointConfigs(nrOfPosCoords);
+                for (size_t i = 0; i < nrOfPosCoords; i++)
+                {
+                    jointConfigs[i] = iDynTree::getRandomDouble();
+                    fullModelPos.jointPos()(fullModel.getJoint(jntIndex)->getPosCoordsOffset() + i) = jointConfigs[i];
+                }
+                removedJointPositions[jointName] = jointConfigs;
+            }
```

Comment:
I don't think this make a lot of sense, probably we can just use `normalizeJointPosCoords` for all joints.